### PR TITLE
 Fix #6546 (2): Proposal for a new structure of the cache list menu

### LINKED ISSUE #6546: Proposed restructuring of the cache list menu
##### Detailed description of the problem:
Adding new functions to the cache list causes a growing of the menu.
Issue [#6528](https://github.com/cgeo/cgeo/issues/6528) asks for a new functionality which adds an additional menu item. In that issue I suggested a restructuring of the main cache list menu.

##### Actual behaviour:
A lot of menu items...

##### Suggestion:
Restructuring of the menu like this:
* Show on map
* Filter
* Sort
* Switch select mode
* Invert select mode
* Manage cashes 
** Refresh all / selected
** Move all / selected
** Copy all / selected
** Remove all / selected 
** Delete offline-logs
** Delete past events
* Manage lists
** Create new list
** Remove current list
** Rename current list
** Make list unique
* Import
** Import GPX
** Import from send2cgeo
** Import from Android
* Export
** Export GPX
** Field notes
** Personal notes
* Remove from history

### PR DESCRIPTION
**Pull request by FUSSEL4**
### Base of the PR
The proposal provided here suggests a new structure of the cache list menu like described in [Issue-6546](https://github.com/cgeo/cgeo/issues/6546). 

This PR is a updated version of [#6551](https://github.com/cgeo/cgeo/pull/6551) which couldn't be rebased/merged due to conflicts.

### Scope of the PR
The scope of the PR is to provide a much shorter main cache list menu by introducing sub menus. 

### Considerations
* A restructuring of the cache list menu seems necessary because new functionality often leads to an additionally menu item (see [#6528](https://github.com/cgeo/cgeo/issues/6528) or  [#6473](https://github.com/cgeo/cgeo/issues/6473)) .
* Two main menu items are added: "Manage caches", "Manage lists".

### Screenshots
| Before | After |
| ----- | ----- |
| device-2017-05-10-230544 | screenshot_20170515-081710 |

### Testing
* Functionality was tested manually (Samsung Galaxy S3 with Kitkat, S6 with Nougat).

### CODE REVIEW

**Comment by Lineflyer** [comment]
OK to test

**Comment by Lineflyer** [comment]
@FUSSEL4 Looks much better now. Hint: You should use the prefix "Fix" or "Fixes" followed be the issue for the PR title to allow autoclosure of the issue. I changed it manually.

**Comment by Lineflyer** [comment]
> The new menu items are written with three dots (&#133;) to symbolize the sub menu which will be opened. This makes sense since only Android N shows a special symbol indicating a sub menu.

I think this comment is outdated, isn't it?
IMHO it is fine without and symbols also on pre-N devices.

File: main/src/cgeo/geocaching/CacheListActivity.java [hunk 1]
```diff
@@ -1,5 +1,38 @@
 package cgeo.geocaching;
 
+import android.app.Activity;
```
Comment (SammysHP) [inline main/src/cgeo/geocaching/CacheListActivity.java]: The imports were re-ordered. I'm not sure what's our current style guideline, but this shouldn't be included in this pull request and fixed in another commit for all files if necessary.

**Comment by SammysHP** [comment]
Oh, and your changes are still based on an old commit (58da41536). You should fetch upstream first and rebase on it.

```
$ git fetch <name-of-upstream-repo>
$ git checkout issue-6546_new_cache_list_menu
$ git rebase <name-of-upstream-repo>/master
```

File: main/src/cgeo/geocaching/CacheListActivity.java (hunk 1 above)
Comment (FUSSEL4) [inline main/src/cgeo/geocaching/CacheListActivity.java]: I restored the previous order.

**Comment by FUSSEL4** [comment]
Can someone rebase this? Obviously, I can't...

**Comment by SammysHP** [comment]
I explained it to you in my previous comment. Now there would be another step to reset to your previous commit.

The one who merges this can do it probably (rebase and squash).

**Comment by FUSSEL4** [comment]
@SammysHP I executed the `fetch`, `checkout` and `rebase` commands like you suggested, and it looked good to me (judging by the success messages on the command line).
Can you please help me with the rebasing? Thanks.

**Comment by pstorch** [comment]
I've rebased and squashed locally and send a new PR. If it is ok, let's close here.
